vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -762,8 +764,14 @@ private void reset() {
             addEmptySelectionItem();
         }
 
+        final AtomicInteger itemCounter = new AtomicInteger(0);
         getDataProvider().fetch(new Query<>()).map(this::createItem)
-                .forEach(this::add);
+                .forEach(component -> {
+                    add(component);
+                    itemCounter.incrementAndGet();
+                });
+
+        lastFetchedDataSize = itemCounter.get();
```

Comment:
This raises now the question that if 2 resets are run at the same time the slower one will get its size set and not the actual order of the calls. (can this happen I can't say, but sounds like we should block the fetch so queries are done in order.

Reply:
Now a dataprovider.fetch() and dataSize update are synchronized on data provider reference.